napari/_qt/qt_viewer.py

```diff
@@ -226,8 +227,26 @@ def screenshot(self):
             upper-left corner of the rendered region.
         """
         img = self.canvas.native.grabFramebuffer()
+        if path is not None:
+            imsave(path, QImg2array(img))  # scikit-image imsave method
         return QImg2array(img)
 
+    def _save_screenshot(self):
+        """Save screenshot of currently displayed screen to file.
+
+        Returns
+        -------
+        filename : str
+            File path of the saved screenshot.
+        """
+        filename, _ = QFileDialog.getSaveFileName(
+            parent=self,
+            caption='',
+            directory=self._last_visited_dir,  # home dir by default
+        )
```

Comment:
I'm also curious here what would it be like to add a checkbox to the defualt `QFileDialog` that allowed people to save the screenshot using the `Viewer.screenshot(..., with_viewer=True)` option when that checkbox is selected. Personally it's quite nice for me to be able to take screenshots with the viewer, but I completely recognize it's a minority use case, so I would make it unchecked by default. But I'm curious if it is possible / how it would feel. We'd also need to move some of the functions around inside this PR to accommodate that, but it shouldn't be too hard. (I could even see a case then for moving `Viewer.screenshot` to `Viewer.window.screenshot` and putting that on the `window` as it is a bit of an outlier on `Viewer`, but this is a separate thread).

Curious what does @tlambert03 think of the checkbox idea?

Reply:
I don't really see the point of adding this feature, I assume anyone who wants to get that end result will just use their system screenshot/clipping tool.

Reply:
It's certainly not essential. Maybe at best it's only mildly convenient for the core team / people that actually want the viewer to appear in their slides. Absolutely fine to ignore, part of what motivated the question was around understanding what it's like to add custom checkboxes / QWidgets to the default save menu as I could imagine a context where we want some additional options for people and maybe don't want another pop-up or entirely custom dialog